### ISSUE TITLE
False alarm in FunctionalInterfaceClash

### ISSUE THREAD

**Issue by PhilippWendler** [google/error-prone#495]
I have an AutoValue builder that extends another class and inherits some abstract setters from the super class. Error-prone 2.0.15 complains about one method in the auto-generated builder class, it says the method is ambiguous if called with lambda parameters. The method with which it collides is the same method from the AutoValue.Builder class. this method also exists in the super class of the AutoValue.Builder class with different return type.

Shortened code:
```
public class Classes {

  public abstract static class ClassLoaderBuilder {

    public abstract ClassLoaderBuilder setUrls(Iterable<URL> urls);

    abstract ExtendedURLClassLoaderConfiguration autoBuild();
  }
}
```

```
  @AutoValue
  abstract static class ExtendedURLClassLoaderConfiguration {
   
    abstract ImmutableList<URL> urls();

    @AutoValue.Builder
    abstract static class AutoBuilder extends Classes.ClassLoaderBuilder {
      AutoBuilder() {}

      @Override
      public abstract AutoBuilder setUrls(Iterable<URL> urls);
    }
  }
```

The error message is
```
    [javac] .apt-generated/org/sosy_lab/common/AutoValue_ExtendedURLClassLoader_ExtendedURLClassLoaderConfiguration.java:107: warning: [FunctionalInterfaceClash] When passing lambda arguments to this function, callers will need a cast to disambiguate with: AutoBuilder.setUrls(Iterable)
    [javac]     public ExtendedURLClassLoader.ExtendedURLClassLoaderConfiguration.AutoBuilder setUrls(Iterable<URL> urls) {
    [javac]                                                                                   ^
    [javac]     (see http://errorprone.info/bugpattern/FunctionalInterfaceClash)
```
The error is in the `setUrls(Iterable<URL> urls)` method of the auto-generated class. The other method it points to is the AutoValue.Builder class AutoBuilder, which specifies `AutoBuilder setUrls(Iterable<URL> urls)`, and which has a super class ClassLoaderBuilder that specifies `ClassLoaderBuilder setUrls(Iterable<URL> urls)`. I suspect error-prone gets confused about overriding methods with different return types here (though it does not mention the super class in the message). IIRC Java adds a synthetic method if I specialize the return type when overriding, maybe this is the problem?

I tried to simplify the code and produce a minimal reproducible example, but when I took either AutoValue or the super class out, the error went away. If you would like to, you could have a look at the real project this code comes from, or I could spent more effort in producing a fully-self contained project for this problem, but I would like to save that effort if you maybe can find the problem otherwise.

**Comment by PhilippWendler** [comment]
I can't reproduce it anymore with error-prone 2.1.1, it seems it is fixed.
